assets/src/js/listing_pages.js

```diff
@@ -7,29 +7,64 @@ export const setupListingPages = () => {
     return;
   }
 
-  // Setup behaviour for list/grid toggle.
-  const listViewToggle = document.querySelector('.list-view-toggle');
-  const gridViewToggle = document.querySelector('.grid-view-toggle');
+  const toggleButton = document.querySelector('.layout-toggle');
 
-  if (!listViewToggle && !gridViewToggle) {
-    return;
-  }
+  const clearStorageAfter = 30 * 60 * 1000; // 30mins in milliseconds
+
+  // Function to clear localStorage after a set time
+  const clearLocalStorage = () => {
+    setTimeout(() => {
+      localStorage.removeItem('layout');
+    }, clearStorageAfter);
+  };
 
-  listingPageContent.classList.toggle('wp-block-query--list', gridViewToggle);
-  listingPageContent.classList.toggle('wp-block-query--grid', listViewToggle);
+  const switchViews = layoutView => {
```

Comment:
Shouldn't we pass `event` as a parameter, too?
```suggestion
  const switchViews = (event, layoutView) => {
```

Reply:
🤔,  yea but _JavaScript_ is smart enough to know the `event` keyword is reserved, also may cause confusion with the `initLayout()` method.